q2_quality_control/quality_control.py

```diff
@@ -19,20 +19,28 @@
 def exclude_seqs(feature_sequences: DNAFASTAFormat,
                  reference_sequences: DNAFASTAFormat, method='blast',
                  perc_identity: float=0.97, evalue: float=0.001,
-                 threads: str=1) -> (pd.Series, pd.Series):
+                 perc_query_aligned: float=0.97, threads: str=1
+                 ) -> (pd.Series, pd.Series):
 
     # BLAST query seqs vs. ref db of contaminants (or targets)
     res = _blast_seqs(
         feature_sequences, reference_sequences, evalue=evalue,
-        perc_identity=perc_identity, threads=threads, method=method)
+        perc_identity=perc_identity, threads=threads,
+        perc_query_aligned=perc_query_aligned, method=method)
 
     # convert feature_sequences to series for filtering
     query_series = _dnafastaformats_to_series(feature_sequences)
 
     # filter seqs from seq file
-    res_md = qiime2.Metadata(res)
-    hits_seqs = filter_seqs(query_series, res_md, exclude_ids=False)
-    misses_seqs = filter_seqs(query_series, res_md, exclude_ids=True)
+    try:
+        res_md = qiime2.Metadata(res)
+        hits_seqs = filter_seqs(query_series, res_md, exclude_ids=False)
+    except ValueError:
+        raise ValueError('No query sequences match the reference.')
```

Comment:
Do we really want the method to fail in these cases? Or should we be creating empty objects? We are able to create empty `FeatureData[Sequence]` objects, as this came up when I was implementing `cluster-features-closed-reference` in `q2-vsearch`.

I realize that it's less important in this case than the closed-reference case (since if you filter no sequences you continue with your input table, and if you feature all... well... you have bigger problems), but it could still be annoying, for example if a user is setting up an automated processing pipeline that includes this filtering process.

Reply:
👍 
Where is this implemented in `cluster-features-closed-reference` though? It looks like a `ValueError` is raised if there are no matches to ref.

Reply:
Either way I see this conflicting with automated pipelines if empty jobs are output, but I agree that may be less of an issue than just failing outright (failing will always cause failure, empty files will only cause failure if they are empty!)

Should we perhaps raise a warning or something to report seq counts?

Reply:
@nbokulich, I was referring to empty [`unmatched_seqs` in `cluster-features-closed-reference`](https://github.com/qiime2/q2-vsearch/blob/de455f66c4c10a0227f116410ae4dfe78222a37c/q2_vsearch/_cluster_features.py#L170), since that result is the same semantic type as the return types here. (No matches to the reference in that case would be an empty feature table, which is something that I think we don't have support for at the moment, and which it makes a little more sense to fail over. Failing because all sequences match the reference would be kind of a bummer...)